extract/extract.py

```diff
@@ -695,11 +688,8 @@ def extract_zip_7zip(self, request: ServiceRequest, local: str, encoding: str, p
         env['LANG'] = 'en_US.UTF-8'
 
         try:
-            stdoutput, stderr = subprocess.Popen(
-                ['7z', 'x', '-p', '-y', local, f'-o{path}'],
-                env=env, stdin=subprocess.PIPE, stdout=subprocess.PIPE,
-                stderr=subprocess.PIPE).communicate()
-            stdoutput += stderr
+            p = subprocess.run(['7z', 'x', '-p', '-y', local, f'-o{path}'], env=env, capture_output=True)
+            stdoutput = p.stdout + p.stderr
```

Comment:
what are the odds that p.stderr will contain "Everything is OK"?

Reply:
Likely none, but the messages like "Wrong Password" or "Can not open the file as archive" could appear in either.

Reply:
ah, I didn't expand the code. Resolving!